projects/framework-photoshop/extensions/plugins/photoshop_document_saved_validator.py

```diff
@@ -0,0 +1,73 @@
+# :coding: utf-8
+# :copyright: Copyright (c) 2024 ftrack
+import os
+
+from ftrack_utils.paths import get_temp_path
+
+from ftrack_framework_core.plugin import BasePlugin
+from ftrack_framework_core.exceptions.plugin import (
+    PluginExecutionError,
+    PluginValidationError,
+)
+
+from ftrack_framework_photoshop.rpc_cep import PhotoshopRPCCEP
+
+
+class PhotoshopSceneSavedValidatorPlugin(BasePlugin):
+    '''
+    Plugin to validate if the Photoshop document has been saved.
+    '''
+
+    name = 'photoshop_document_saved_validator'
+
+    def save_document_to_temp(self, store):
+        '''
+        Save the current Photoshop document.
+        '''
+
+        # Get existing RPC connection instance
+        photoshop_connection = PhotoshopRPCCEP.instance()
+
+        self.logger.warning('Photoshop document not saved, asking to save')
+        temp_path = get_temp_path('psd')
```

Comment:
If we want to support multiple filetypes (psd and psb for example) and be 100% aligned, extension_format should come from the collector.